docs/introduction/javascript-events-dom-apis.md

```diff
@@ -502,6 +502,41 @@ entityEl.addEventListener('physicscollided', collisionHandler);
 entityEl.removeEventListener('physicscollided', collisionHandler);
 ```
 
+### Binding Event Listeners
+
+By default, [Javascript execution context rules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) binds `this` to the global context (`window`) for any independent function, meaning that these functions won't have access to the component's `this` by default.
+
+In order for the component's `this` to be accessible inside an event listener, [it must be bound](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this).
+
+There are several ways you can do this:
+
+1. By using an arrow function to define the event listener.  Arrow functions automatically bind `this`.
+
+```entityEl.addEventListener('physicscollided', function (event) {
+entityEl.addEventListener('click', (event) => {
+  console.log(this.el.id);
+});
+```
+
+2. By defining your event listener within the events object of the component (this will also handling adding and removing the listener automatically)
+
+   See the explanation [here](../core/component.html#events).
+
+3. By binding the function as you add the event listener.  You will also need to do the same when [removing the event listener.](#removing-an-event-listener-with-removeeventlistener)
```

Comment:
Are 2, 3 and 4 redundant? `explicit binding`

Reply:
I don't think 2 is redundant.  It's a good solution - probably the most efficient overall (i.e. least code to write), but because it obscures what's going on under the covers, I didn't think it should be the only solution offered.

3 & 4 are both approaches I have seen used very widely, but arguably there is very little difference between them.  I'd be quite happy to drop one or other of these.  I presonally usually use 3 as it's more compact, but have seeen a lot of examples that use 4.

I think it is good to show both `bind(this)` and arrow functions.

Reply:
Yeah 3 seems too specific to a use case. I would remove. Thanks

Reply:
Removed #3 and renumbered # 4 to #3.